### ISSUE TITLE
rtl is not displaying the items correctly (or none at all)

### ISSUE THREAD

**Issue by sag1v** [akiran/react-slick#952]
I'm facing same issues as mentioned in #83 and  #926  and #880  and #839  .
It seems that the `tranform: translate3d` and `width` calculations of `.slick-track are off`.
so when `rtl` is set to true, you can't see the slides.
If i disable the inline style of `translate3d` and `width` it shows all the items.

After playing with the source code a bit (in `src/mixins/trackHelper`),
I think the solution or at least part of it, is to change [this block code](https://github.com/akiran/react-slick/blob/master/src/mixins/trackHelper.js#L20):
```
 if (!spec.vertical) {
    if (spec.variableWidth) {
      trackWidth = (spec.slideCount + 2*spec.slidesToShow) * spec.slideWidth;
    } else if (spec.centerMode) {
      trackWidth = (spec.slideCount + 2*(spec.slidesToShow + 1)) * spec.slideWidth;
    } else {
      trackWidth = (spec.slideCount + 2*spec.slidesToShow) * spec.slideWidth;
    }
  } else {
    trackHeight = trackChildren * spec.slideHeight;
  }
```

To this:
```
  if (!spec.vertical) {
	if(spec.rtl){ // added condition for rtl
		trackWidth = 0;
	} else if (spec.variableWidth) {
      trackWidth = (spec.slideCount + 2 * spec.slidesToShow) * spec.slideWidth;
    } else if (spec.centerMode) {
      trackWidth = (spec.slideCount + 2 * (spec.slidesToShow + 1)) * spec.slideWidth;
    } else {
      trackWidth = (spec.slideCount + 2 * spec.slidesToShow) * spec.slideWidth;
    }
  } else {
    trackHeight = trackChildren * spec.slideHeight;
  }
```
and[ this block code](https://github.com/akiran/react-slick/blob/master/src/mixins/trackHelper.js#L32):
```
  var style = {
    opacity: 1,
    WebkitTransform: !spec.vertical ? 'translate3d(' + spec.left + 'px, 0px, 0px)' : 'translate3d(0px, ' + spec.left + 'px, 0px)',
    transform: !spec.vertical ? 'translate3d(' + spec.left + 'px, 0px, 0px)' : 'translate3d(0px, ' + spec.left + 'px, 0px)',
    transition: '',
    WebkitTransition: '',
    msTransform: !spec.vertical ? 'translateX(' + spec.left + 'px)' : 'translateY(' + spec.left + 'px)',
  };
```
to this:
```
  var style = {
    opacity: 1,
    WebkitTransform: !spec.vertical ? 'translate3d(' + (spec.rtl? 0 : spec.left) + 'px, 0px, 0px)' : 'translate3d(0px, ' + spec.left + 'px, 0px)',
    transform: !spec.vertical ? 'translate3d(' + (spec.rtl? 0 : spec.left) + 'px, 0px, 0px)' : 'translate3d(0px, ' + spec.left + 'px, 0px)',
    transition: '',
    WebkitTransition: '',
    msTransform: !spec.vertical ? 'translateX(' + (spec.rtl? 0 : spec.left) + 'px)' : 'translateY(' + spec.left + 'px)'
  };
```

It would be nice if others can confirm that this solution works well.

**Comment by sag1v** [comment]
@akiran Do you know about this issue?

**Comment by laveesingh** [comment]
@sag1v could you give your configuration/code-snippet that displays this behaviour?
I tried some on my own but didn't encounter it. Although I faced a problem with swiping direction in case of {rtl: true} and that'll be fixed soon. 
Please feel free to reopen it, if you can suggest to regenerate it.

**Comment by sag1v** [comment]
@laveesingh Sorry for not replying earlier.  
You can reproduce this bug with a simple code:  
```
const images = [
  "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcROCs4gOdKWhzHyqoq2M5ysh57yLXpc77fKwAdTz8CJnPXBV-nI7A",
  "https://i.ytimg.com/vi/RHLknisJ-Sg/maxresdefault.jpg",
  "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQZ99KJVZdeRSgw2kJkGcX3_Yf7ZWsHZJx38bf_lLomX2lApFtd",
]

class App extends React.Component {
  render() {
    return (
      <div style={{direction: 'rtl'}}>
        <Slider
          rtl={true}
          draggable={false}
        >
          {
            images.map((image, i) => <img key={i} src={image} />)
          }
        </Slider>
      </div>
    );
  }
}

render(<App />, document.getElementById('root'));
```

Note that the bug is there when you set the css style `direction:rtl`.  
Anyway here is a [Sandbox snippet](https://codesandbox.io/embed/k3q28w7p9v)

**Comment by laveesingh** [comment]
This simple snippet worked perfectly for me:
```    return (
      <div>
        <Slider rtl={true} draggable={false} autoplay={true}>
          {images.map((image, i) => <img key={i} src={image} />)}
        </Slider>
      </div>
    );
```
The culprit here seems to be `style={{direction: 'rtl'}}` in wrapper div.

**Comment by sag1v** [comment]
@laveesingh 
>The culprit here seems to be style={{direction: 'rtl'}} in wrapper div.

Indeed, that's exactly what i wrote. And this will be the case most of the time because when a site / app needs to display right to left markup it will have somewhere in the root this styling:
`style={{direction: 'rtl'}}`

**Comment by jraut** [comment]
Languages which are written _and_ arranged/composed left to right do require the direction to be set ltr, either by setting html attribute dir or giving a css style attribute direction. This fix (along with https://github.com/akiran/react-slick/pull/1069) focuses on enabling this component to be used in pages which actually use rtl.

tl;dr: `style={{direction: 'rtl'}}` is actually neccessary.

**Comment by sag1v** [comment]
I'm using my [own version of `react-slick`](https://github.com/sag1v/react-slick) which has a small fix for the `rtl`. i would love to move back to the original master as soon as this issue will get resolved.

**Comment by laveesingh** [comment]
I have found a fix for this: 6395faa52f908b9e023b51a77ef3e38f0d5300fb

**Comment by mcfoton** [comment]
so was this fixed? or is there a working workaround to use for now?

**Comment by sag1v** [comment]
Was this issue fixed and released yet?

**Comment by mcfoton** [comment]
@akiran can you please update on this? do you need any clarifications\demo on the exact issue?

**Comment by sag1v** [comment]
It's not only the initial load, i think any calculation taking place when `rtl` is set to true are off.
for example `afterChange` gives a wrong `index`.

**Comment by hrkazemi** [comment]
why this issue closed?

**Comment by protowassim** [comment]
This issue is still not fixed.

**Comment by hrkazemi** [comment]
> This issue is still not fixed.

No, I don’t believe the issue is rectified,
What you can do is to reverse the slide items list and also
set "initialSlide" value to  "slide_items.legnth - slidesToShow" 
Otherwise, some item gaps will appear at the beginning.

**Comment by sag1v** [comment]
The bug of this feature and another one were the main reasons i decided to create my own carousel library    🤷🏻‍♂️

**Comment by protowassim** [comment]
Can you share your library and steps on how to include it in the project? I
assume it needs to override the default react-slick and will have to
imported in all pages required by this module.

**Comment by sag1v** [comment]
Yeah sure 
https://github.com/sag1v/react-elastic-carousel

**Comment by mooSylla** [comment]
Just ran into this bug today! thanks @sag1v for the workaround.
